/**
 * Copyright 2019 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

package mindspore.straspb;

message ParallelStrategy {
    repeated uint32 dim = 1;
}

message ParallelStrategys {
    required uint32 stage = 1;
    repeated ParallelStrategy parallel_strategy = 2;
}

message ParallelStrategyItem {
    required string node_name = 1;
    required ParallelStrategys parallel_strategys = 2;
}

message ParallelStrategyMap {
    required uint32 train_time = 1;
    repeated ParallelStrategyItem parallel_strategy_item = 2;
}